packages/@aws-cdk/aws-apigatewayv2/test/http/api.test.ts

```diff
@@ -531,6 +532,63 @@ describe('HttpApi', () => {
   });
 });
 
+describe('SpecHttpApi', () => {
+  test('default', () => {
+    const stack = new Stack();
+    const oas = {
+      openapi: '3.0.2',
+      paths: {
+        '/pets': {
+          get: {
+            'responses': {
+              200: {
+                content: {
+                  'application/json': {
+                    schema: {
+                      $ref: '#/components/schemas/Empty',
+                    },
+                  },
+                },
+              },
+            },
+            'x-amazon-apigateway-integration': {
+              responses: {
+                default: {
+                  statusCode: '200',
+                },
+              },
+              requestTemplates: {
+                'application/json': '{"statusCode": 200}',
+              },
+              passthroughBehavior: 'when_no_match',
+              type: 'mock',
+            },
+          },
+        },
+      },
+      components: {
+        schemas: {
+          Empty: {
+            title: 'Empty Schema',
+            type: 'object',
+          },
+        },
+      },
+    };
+    const api = new SpecHttpApi(stack, 'api', {
+      apiDefinition: ApiDefinition.fromInline(oas),
+    });
+
+    Template.fromStack(stack).hasResourceProperties('AWS::ApiGatewayV2::Api', {
+      Body: oas,
+    });
+
+    expect(api.apiEndpoint).toBeDefined();
+    expect(api.apiId).toBeDefined();
+    expect(api.httpApiId).toBeDefined();
```

Comment:
These should be tested against the expected output, not just that they're defined.